packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -577,6 +613,16 @@ export class Pipeline extends PipelineBase {
 
   private getRoleFromActionPropsOrGenerateIfCrossAccount(stage: Stage, action: IAction): iam.IRole | undefined {
     const pipelineStack = Stack.of(this);
+    const actionEnv = new RichAction(action).env;
+
+    // Do a check on whether the cross-account acccess is possible given the
+    // current pipeline configuration. This has bearing on whether
+    // `artifactBucket.grantRead(action.role)` (which is going to occur down the
+    // line) can even work.
+    const hasAccount = !Token.isUnresolved(actionEnv.account);
+    if (hasAccount && !sameEnvDimension(this.env.account, actionEnv.account) && !this.crossAccountKeys) {
+      throw new Error(`Pipeline must be created with 'crossAccountKeys: true' to add cross-account action '${action.actionProperties.actionName}'. Pipeline account: '${renderEnvDimension(this.env.account)}', action account: '${renderEnvDimension(actionEnv.account)}'`);
```

Comment:
This line is 263 characters long... did we remove that liner rule?

Please break the string into multiple lines here.

Reply:
It's because it's a string that the rule doesn't fire. Strings don't hide complex conditionals you might otherwise miss.